lib/AST/Type.cpp

```diff
@@ -2517,7 +2517,17 @@ TypeSubstitutionMap TypeBase::getMemberSubstitutions(DeclContext *dc) {
     }
 
     // Continue looking into the parent.
-    baseTy = baseTy->castTo<NominalType>()->getParent();
+    auto nominalTy = baseTy->getAs<NominalType>();
+    if (!nominalTy || nominalTy->is<ErrorType>()) {
```

Comment:
The getAs will either produce a NominalType or null; there's no need to check for ErrorType here.